src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,32 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.NotEnoughAgeException;
+import core.basesyntax.exceptions.NotEnoughSizeException;
+import core.basesyntax.exceptions.UserIsNullException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new UserIsNullException("Can't register user, because user is null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            return null;
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new NotEnoughSizeException("The length of the login must be at least 6");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new NotEnoughSizeException("The length of the password must be at least 6");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new NotEnoughAgeException("The age of the user must be at least 18");
+        }
```

Comment:
[Try to avoid these common mistakes, while solving task](https://mate-academy.github.io/jv-program-common-mistakes/java-core/junit/registration-validation)

![image](https://github.com/mate-academy/jv-registration-validation/assets/113326347/d2041d83-18df-4972-9501-61b764502888)